src/game_api/entities_items.hpp

```diff
@@ -289,6 +289,12 @@ class Spark : public Flame
     float next_size;
     /// very short timer before next size change, giving a pulsing effect
     uint8_t size_change_timer;
+    uint8_t extra_padding[3];
+
+    /// This is cusome variable, you need to use `activate_sparktraps_hack` to use it
+    float speed;
+    /// This is cusome variable, you need to use `activate_sparktraps_hack` to use it
+    float distance;
```

Comment:
This is a bit of a yolo, but I guess considering how the game allocates entities it's fine  😆 
Have you just double-checked that the size of a sparktrap is not right at the border of the bucket size?

Reply:
Yeah, we have quite a bit of room in fact